### ISSUE TITLE
KDE Invent support

### ISSUE THREAD

**Issue by BernardoGomesNegri** [hickford/git-credential-oauth#47]
[KDE Invent](https://invent.kde.org) is the gitlab based git forge used by KDE, the group responsible for developing the Plasma Desktop Environment for Linux as well as numerous applications for it.

Unfortunately, it offers no way for users to register their own OAuth apps, so it would be great if git-credential-oauth talked to them and registered an instance wide OAuth application.

**Comment by hickford** [comment]
Is there a mailing list or contact information for the admins of https://invent.kde.org/ ?

**Comment by BernardoGomesNegri** [comment]
https://discuss.kde.org/ should do I think

**Comment by hickford** [comment]
Does this work?

    git config --global credential.https://invent.kde.org.oauthclientid 36ee2eeb4f179b38db8f066fd4ca9751f4b003653a205fc1422f110ca60181f6

**Comment by BernardoGomesNegri** [comment]
No, git credential-oauth does not even appear to try and generate a URL:
```
[bernardo@computer project] GIT_TRACE=1 git push origin test-branch
13:08:55.909354 git.c:463               trace: built-in: git push origin test-branch
13:08:55.912276 run-command.c:657       trace: run_command: GIT_DIR=.git git remote-https origin https://invent.kde.org/bernardogn/project
13:08:55.913935 git.c:749               trace: exec: git-remote-https origin https://invent.kde.org/bernardogn/project
13:08:55.913971 run-command.c:657       trace: run_command: git-remote-https origin https://invent.kde.org/bernardogn/project
13:08:56.901160 run-command.c:657       trace: run_command: 'git credential-cache --timeout 7200 get'
13:08:56.904713 git.c:463               trace: built-in: git credential-cache --timeout 7200 get
13:08:56.905047 run-command.c:657       trace: run_command: 'git credential-oauth get'
13:08:56.908155 git.c:749               trace: exec: git-credential-oauth get
13:08:56.908193 run-command.c:657       trace: run_command: git-credential-oauth get
13:08:56.928326 run-command.c:657       trace: run_command: /usr/bin/ksshaskpass 'Username for '\''https://invent.kde.org'\'': '
[bernardo@computer project] printf 'protocol=https\nhost=invent.kde.org\nwwwauth=Bearer realm=Gitlab\n' | git credential-oauth get
# No output
[bernardo@computer project] git config credential.helper
oauth
```

**Comment by hickford** [comment]
How about this?

```
git config --global credential.https://invent.kde.org.oauthclientid 36ee2eeb4f179b38db8f066fd4ca9751f4b003653a205fc1422f110ca60181f6
git config --global credential.https://invent.kde.org.oauthAuthURL https://invent.kde.org/oauth/authorize
git config --global credential.https://invent.kde.org.oauthTokenURL https://invent.kde.org/oauth/token
git config --global credential.https://invent.kde.org.oauthScopes "read_repository write_repository"
```

**Comment by BernardoGomesNegri** [comment]
That works fine! Thank you very much

**Comment by hickford** [comment]
Fixed in v0.11.1 https://github.com/hickford/git-credential-oauth/releases/tag/v0.11.1
